File: TestLua/first.lua

-- Делаем массив случайных чисел от 1 до 10
-- Нечётные числа умножаем на 2
-- Чётные числа делим на 2
-- Числа меньшие n вывести на экран

-- Вау, Lua

-- Переменные
local a = {}

-- Ввод размера массива
print("Enter array size:")
local d = io.read()

-- Изначальный массив
print("Our array:")
for i = 1, d do
    a[i] = math.random(1,10)
    print(a[i])
end

-- Изменённый массив
print("Changed array:")
for i = 1, d do
    if math.fmod(a[i], 2) == 0 then
        a[i] = a[i] / 2
    else
        a[i] = a[i] * 2
    end
    a[i] = math.floor(a[i])
    print(a[i])
end

-- Ввод числа n
print("Enter n:")
local n = io.read()
n = tonumber(n)

-- Числа меньше n
print("Less than n:")
for i = 1, d do
    if a[i] < n then
    print(a[i])
    end
end

File: TestPy/first.py

# Делаем массив случайных чисел от 1 до 10
# Нечётные числа умножаем на 2
# Чётные числа делим на 2
# Числа меньшие n вывести на экран

# Вау, Python

import random, math

a = {}
d = 5
n = 5
i = 0

# Изначальный список
print("Начальный массив:")
while i < d:
    a[i] = random.randint(1, 10)
    print(a[i])
    i += 1
i = 0

# Изменённый список
print("Изменённый массив:")
while i < d:
    if math.fmod(a[i], 2) == 0:
        a[i] /= 2
    else:
        a[i] *= 2
    print(a[i])
    i += 1
i = 0

# Числа меньше n

print("меньше n:")
while i < d:
    if a[i] < n:
        print(a[i])
    i += 1
i = 0